### PR TITLE
improve-navigation-#16. Change navigation to be AppBar instead of BottomNavigation

### PR DESCRIPTION
**Pull request by andersnylund**
I thought that this would be more intuitive and follow basic rules of navigation. What do you think? If this is OK I will continue improving the navigation based on this. The color will probably change. There could also be a logo all the way to the left

### CODE REVIEW

**Comment by andersnylund** [comment]
Also, what do you think about the usage of styled-components in this component? I personally find this a lot more readable than what can be found from material-ui's examples. IIRC the guideline with styled-components is that `className` on components should never be used

File: src/charts/ChartContainer.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+const Div = styled.div`
+  margin: 6rem auto;
+  width: 60rem;
```
Comment (mansour-ahmed) [inline src/charts/ChartContainer.js]: I recommend here adding `max-width: 60rem` and then `width:100%` to have it responsive (if the library supports responsive charts)

File: src/components/NavigationAppBar.js [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    const renderMobileMenu = (
+      <Menu
+        anchorEl={mobileMoreAnchorEl}
+        anchorOrigin={{ vertical: 'top', horizontal: 'right' }}
+        transformOrigin={{ vertical: 'top', horizontal: 'right' }}
+        open={isMobileMenuOpen}
+        onClose={this.handleMobileMenuClose}
+      >
+        <MenuItem onClick={() => this.handleMenuClick(0)}>
+          <IconButton color="inherit">
```
Comment (tu-tran) [inline src/components/NavigationAppBar.js]: ToolTip would be nice

File: src/components/NavigationAppBar.js [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+  width: 100%;
+`;
+
+const Grow = styled.div`
+  flex-grow: 1;
+`;
+
+const SectionDesktop = styled.div`
+  display: none;
+  /* @TODO define what width mobile is */
```
Comment (mansour-ahmed) [inline src/components/NavigationAppBar.js]: isn't there a default value from material for that?

File: src/components/NavigationAppBar.js [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+`;
+
+const SectionMobile = styled.div`
+  display: flex;
+  @media (min-width: 60rem) {
+    display: none;
+  }
+`;
+
+const Title = styled.div`
```
Comment (mansour-ahmed) [inline src/components/NavigationAppBar.js]: are we sure we want to hide this?

File: src/components/NavigationAppBar.js [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+
+    const renderMobileMenu = (
+      <Menu
+        anchorEl={mobileMoreAnchorEl}
+        anchorOrigin={{ vertical: 'top', horizontal: 'right' }}
+        transformOrigin={{ vertical: 'top', horizontal: 'right' }}
+        open={isMobileMenuOpen}
+        onClose={this.handleMobileMenuClose}
+      >
+        <MenuItem onClick={() => this.handleMenuClick(0)}>
```
Comment (mansour-ahmed) [inline src/components/NavigationAppBar.js]: would be nice to have the active item highlighted

Comment (mansour-ahmed) [inline src/components/NavigationAppBar.js]: I think there is an opportunity to refactor this MenuItem into an own component, you can decide if you would like to do it now or add a todo and refactor it later.

File: src/pages/VisualizationPage.js [hunk 6]
```diff
@@ -1,36 +1,60 @@
...
+  state = { value: 0 };
+
+  handleChange = (event, value) => {
+    this.setState({ value });
+  };
+
+  getChart = () => {
+    const { initialTransactions, categories } = this.props;
+    const { value } = this.state;
+    if (value === 0) return <TransactionHistory initialTransactions={initialTransactions} />;
```
Comment (mansour-ahmed) [inline src/pages/VisualizationPage.js]: Suggestion: I would assign the numbers to a const variables and then do the comparison against the variable instead to improve readability of the code and the ease of modifying  of different steps.

**Comment by andersnylund** [comment]
> Hmm I thought there is already some "background" for selected page?

No, I think that your touch display was playing with us as there is not logic looking at what page is selected.

**Comment by mansour-ahmed** [comment]
This is looking good now, one minor thing that is missing is showing the header's title on both mobile and desktop but thats not a blocker to block merging this PR. in case @hostmax has no further comments i think we should merge this!

**Pull request merged**